### PR TITLE
fix(workflow): Query postgres for first/last release instead of snuba

### PR DESCRIPTION
**Pull request by iProgramStuff**
Currently, tagstore.__get_release queries against snuba to get the first or last release of a group. This information is available in Postgres and can be fetched more quickly. This PR switches _get_release over to using Postgres.

Example transaction today showing the snuba query taking 240 ms: https://sentry.io/organizations/sentry/performance/sentry:c1e8b1376f484c0daaaf484632bcd534/

Query getting the same data from postgres taking less time: https://gist.github.com/JTCunning/90eccd2bc1cedeb8bf8086917c5a4661

Resolves WOR-640

### CODE REVIEW

File: src/sentry/tagstore/snuba/backend.py
Comment (wedamija) [inline src/sentry/tagstore/snuba/backend.py:543]: This will cause an extra query. Let's just do 
```
group_releases = list(GroupRelease.objects.filter(
            group_id=group_id,
            project_id=project_id,
).select_related('release').order_by(orderby)[:1])
```

File: src/sentry/tagstore/snuba/backend.py
Comment (wedamija) [inline src/sentry/tagstore/snuba/backend.py:536]: I don't think this belongs in tagstore anymore, since we're no longer querying Snuba here. Can we move this into `group.get_(first|last)_release`?

Nice thing here is that we can change the repair query in `get_first_release` to just use this query as well.

**Review by wedamija — COMMENTED** [review]
We'll need to add indexes to `GroupRelease`:
 - `group, release, first_seen`
 - `group, release, last_seen`

**Comment by wedamija** [comment]
> We'll need to add indexes to `GroupRelease`:
> 
> * `group, release, first_seen`
> * `group, release, last_seen`

Oops, this should have been `group, projects, ...` instead of release, since we aren't querying on release.

**Comment by iProgramStuff** [comment]
> > We'll need to add indexes to `GroupRelease`:
> > 
> > * `group, release, first_seen`
> > * `group, release, last_seen`
> 
> Oops, this should have been `group, project, ...` instead of release, since we aren't querying on release.
> 
> [Edit] Thinking about this even more, wondering if `group_id, first_seen` and `group_id, last_seen` are enough. Do we actually even need to query by project id here?

I don't really think so. I don't imagine there are entries for group + release combinations that have different projects.

File: src/sentry/models/group.py
Comment (wedamija) [inline src/sentry/models/group.py:495]: To fit this into one query, maybe we could just not convert `group_releases` to a list and use it as a subquery instead. So we could use 
```
Release.objects.get(
    id__in=GroupRelease.objects.filter(group_id=group_id, project_id=project_id,).order_by(orderby).values("release_id")[:1]
)
```

Then we can just catch `Release.DoesNotExist` and skip the extra checks as well. Probably worth just testing this out and posting the sql here so that we can verify it makes sense.

Comment (iProgramStuff) [reply src/sentry/models/group.py:495]: This is great Dan.

**Review by wedamija — COMMENTED** [review]
LGTM, just want to wait for approval until the index is created

**Review by wedamija — APPROVED** [review]

File: src/sentry/models/group.py
Comment (beezz) [inline src/sentry/models/group.py:486]: Should this be `id__in` if we are comparing it to a slice `[:1]`?

File: src/sentry/models/group.py
Comment (wedamija) [inline src/sentry/models/group.py:497]: Set this to `False`, then you can do `if release_version is not None` above. That way we cache the negative lookup.

**Review by wedamija — CHANGES_REQUESTED** [review]
Just removing approval until we get the cache stuff sorted out

File: src/sentry/models/group.py
Comment (wedamija) [inline src/sentry/models/group.py:488]: Might slightly safer here to have an `else` that sets the `release_version` to `None` when it's `False`

**Review by wedamija — APPROVED** [review]

**Pull request merged**
